src/pkg/bundle/dev.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The UDS Authors
+
+// Package bundle contains functions for interacting with, managing and deploying UDS packages
+package bundle
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/defenseunicorns/uds-cli/src/config"
+	zarfCLI "github.com/defenseunicorns/zarf/src/cmd"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	zarfUtils "github.com/defenseunicorns/zarf/src/pkg/utils"
+)
+
+// CreateZarfPkg creates a zarf package if its missing when in dev mode
+func (b *Bundle) CreateZarfPkgs(srcDir string) {
+	path := filepath.Join(srcDir, b.cfg.CreateOpts.BundleFile)
+	if err := zarfUtils.ReadYaml(path, &b.bundle); err != nil {
+		message.Fatalf(err, "Failed to read bundle.yaml: %s", err.Error())
+	}
+
+	zarfPackagePattern := `^zarf-.*\.tar\.zst$`
+	for _, pkg := range b.bundle.Packages {
+		if pkg.Repository == "" {
+			path := srcDir + pkg.Path
+			// get files in directory
+			files, err := os.ReadDir(path)
+			if err != nil {
+				message.Fatalf(err, "Failed to obtain package %s: %s", pkg.Name, err.Error())
+			}
+			regex := regexp.MustCompile(zarfPackagePattern)
+
+			// check if package exists
+			packageFound := false
+			for _, file := range files {
+				if regex.MatchString(file.Name()) {
+					packageFound = true
+					break
+				}
+			}
+			// create local zarf package if it doesn't exist
+			if !packageFound {
+				os.Args = []string{"zarf", "package", "create", path, "--confirm", "-o", path}
+				zarfCLI.Execute()
```

Comment:
Instead of using the Zarf CLI directly, any idea what it looks like to use the Zarf lib?

Reply:
so the lib route ends up looking like this:
```
...
pkgConfig := types.PackagerConfig{}
pkgConfig.CreateOpts.BaseDir = path
pkgConfig.CreateOpts.Output = path

var isCleanPathRegex = regexp.MustCompile(`^[a-zA-Z0-9\_\-\/\.\~\\:]+$`)
if !isCleanPathRegex.MatchString(config.CommonOptions.CachePath) {
	message.Warnf(lang.CmdPackageCreateCleanPathErr, zarfConfig.ZarfDefaultCachePath)
	config.CommonOptions.CachePath = zarfConfig.ZarfDefaultCachePath
}

// Ensure uppercase keys from viper
v := common.GetViper()
pkgConfig.CreateOpts.SetVariables = helpers.TransformAndMergeMap(
	v.GetStringMapString(common.VPkgCreateSet), pkgConfig.CreateOpts.SetVariables, strings.ToUpper)

// Configure the packager
pkgClient := packager.NewOrDie(&pkgConfig)
defer pkgClient.ClearTempPaths()

// Create the package
if err := pkgClient.Create(); err != nil {
	message.Fatalf(err, lang.CmdPackageCreateErr, err.Error())
}
```
I agree the cli route is a little weird, but it is less code/cleaner which is nice

Reply:
👍 cli route sounds good to me!